### DISCUSSION TITLE
Specifying canister init arguments in dfx.json

### DISCUSSION THREAD

**Discussion by ocluf** [dfinity/sdk#2887] [Feature Requests]
It would be nice if you could could specify the init arguments for the canister in your dfx.json. 

For example in the current [token deployment tutorial](https://forum.dfinity.org/t/confusion-about-documentation-whats-the-purpose-of-deploy-new-token-vs-dip20-etc/16296) 

You have to copy this command which you would probably put in a shell script.

```
dfx deploy --network ${NETWORK} custom-ledger --argument '(record {
  token_name = opt "'${TOKEN_NAME}'";
  token_symbol = opt "'${TOKEN_SYMBOL}'";
  minting_account = "'${MINT_ACC}'";
  initial_values = vec {};
  send_whitelist = vec {};
  archive_options = opt record {
    trigger_threshold = 2000;
    num_blocks_to_archive = 1000;
    controller_id = principal "'${ARCHIVE_CONTROLLER}'";
    cycles_for_archive_creation = opt 10_000_000_000_000
  }
})'
```

While this isn't a huge bother it would be a nice quality of life improvement if you could just specify these args in the dfx.json so all your canister configuration is one place. Then you also could use `dfx deploy --all` with different init arguments per canister. 

It would also be nice if you could make a distinction between the arguments that are passed locally and when you deploy to the IC. In the token example you could for instance pass different minting principals for local en network deployment.

Perhaps this was hinted at in https://github.com/dfinity/sdk/discussions/2632 but it didn't mention dfx.json so I created a new discussion.

**Comment by ghost** [comment]
This is implemented in https://github.com/dfinity/sdk/pull/3571 for release in dfx 0.17.0
